contents/docs/error-tracking/_snippets/nextjs-install-error-tracking.mdx

```diff
@@ -117,4 +117,58 @@ export const onRequestError = async (err, request, context) => {
 }
 ```
 
-You can find a full example of both this and client-side error tracking in our [Next.js error monitoring tutorial](/tutorials/nextjs-error-monitoring).
\ No newline at end of file
+You can find a full example of both this and client-side error tracking in our [Next.js error monitoring tutorial](/tutorials/nextjs-error-monitoring).
+
+## Uploading source maps
+
+import CLIDownload from "./cli/download.mdx"
+import CLIAuthenticate from "./cli/authenticate.mdx"
+import CLIInject from "./cli/inject.mdx"
+import CLIUpload from "./cli/upload.mdx"
+
+If your source maps are not publicly hosted, you will need to upload them during your build process to see unminified code in your stack traces. The `posthog-cli` handles this process.
+
+### Download the CLI
+
+<CLIDownload/>
+
+### Authenticate the CLI
+
+<CLIAuthenticate />
+
+### Generate source maps
+
+Next.js disables source maps by default during production builds to prevent you from leaking your source on the client. You can opt-in with the `productionBrowserSourceMaps` configuration flag.
+
+We will set a environment variable to control this so we only generate source maps when building locally (or if you are running a build in CI).
+
+```file=.env.local
+NEXT_PUBLIC_POSTHOG_SOURCE_MAPS=true
+```
+
+Once we have the environment variable set, we can configure Next.js to generate source maps.
+
+```js file=next.config.js 
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  productionBrowserSourceMaps: process.env.GENERATE_SOURCEMAPS === 'true'
```

Comment:
Is this meant to be `NEXT_PUBLIC_POSTHOG_SOURCE_MAPS`? Not sure if I'm misunderstanding this or is a typo

Reply:
Oops, typo on the above name, good catch!